test/controllers/sections/items/ItemProducerSizeControllerSpec.scala

```diff
@@ -0,0 +1,212 @@
+    def onSubmitAction(idx: Index): Call = routes.ItemProducerSizeController.onSubmit(testErn, testDraftId, idx, NormalMode)
+  }
+
+
+  "ItemProducerSize Controller" - {
+
+    "for a GET onPageLoad" - {
+
+      "must return OK and the correct view for a GET with a goodstype set" in new Setup(Some(
+        emptyUserAnswers.set(ItemExciseProductCodePage(0), Wine.code)
+      )) {
+
+        val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form,
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2022",
+          endYear = "2023"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must return OK and the correct view for a GET when the current date is January" in new Setup(
+        userAnswers = Some(emptyUserAnswers.set(ItemExciseProductCodePage(0), Wine.code)),
+        monthValue = 1
+      ) {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form,
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2021",
+          endYear = "2022"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must populate the view correctly on a GET when the question has previously been answered" in new Setup(Some(
+        emptyUserAnswers
+          .set(ItemExciseProductCodePage(0), Wine.code)
+          .set(ItemProducerSizePage(0), BigInt(1))
+      )) {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form.fill(1),
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2022",
+          endYear = "2023"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must redirect to ItemsIndexController for a GET there is no GoodsType in UserAnswers" in new Setup() {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.sections.items.routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+      }
+
```

Comment:
Could you add the test for index out of bounds? Should be a copy-paste from other controller tests